pytest-embedded-idf/pytest_embedded_idf/dut.py

```diff
@@ -127,3 +129,88 @@ def close(self) -> None:
         if not self.skip_check_coredump:
             self._check_coredump()
         super().close()
+
+    Case = namedtuple('Case', 'index name type keyword group attribute subcase')
+
+    def get_test_menu(
+        self, regex: str = "Here's the test menu, pick your combo:(.+)Enter test for running.", trigger: str = ''
+    ) -> List[Case]:
+        """
+        Get test case list from test menu via UART print.
+
+        This function should be called after `Press ENTER to see the list of tests` prompt is shown.
```

Comment:
How about moving this expect statement into this function as well, as an argument with a default value.

otherwise, a common test script would be 

```python
dut.expect('Press ENTER to see the list of tests')
menu_cases = dut.get_test_menu()
```

which is a bit redundant.

Reply:
Test menu can be printed anytime when the device is idle, by pressing ENTER. So if we move expection of the prompt inside this function, we can only call this function right after test app launches. In fact, `Press ENTER to see the list of tests` is more like a signal of "device is ready" than "menu print is ready", so I think it's better to leave it in the top level of the test app to indicate it is ready for users to execute tests.

Reply:
For me, passing this as an argument with a default value is the same as giving the `trigger` as an argument with a default value.

Or you could split it up into two functions. One is `parse_unity_menu_from_str(s: str)`, which is a `staticmethod` that would parse the menu from a string. The other one is the current `parse_unity_menu`, which would have the `ready line`, `trigger`, and the `end pattern`.

By splitting it up into two functions, it would be easier to write a small test function without interacting with an actual test device. like https://github.com/espressif/pytest-embedded/blob/main/pytest-embedded/tests/test_base.py#L290